src/stan/mcmc/hmc/hamiltonians/dense_e_point.hpp

```diff
@@ -11,29 +11,54 @@ namespace mcmc {
  * Euclidean manifold with dense metric
  */
 class dense_e_point : public ps_point {
- public:
+private:
   /**
    * Inverse mass matrix.
    */
   Eigen::MatrixXd inv_e_metric_;
+  Eigen::MatrixXd inv_e_metric_llt_matrixU_;
+public:
 
   /**
    * Construct a dense point in n-dimensional phase space
    * with identity matrix as inverse mass matrix.
    *
    * @param n number of dimensions
    */
-  explicit dense_e_point(int n) : ps_point(n), inv_e_metric_(n, n) {
+  explicit dense_e_point(int n) : ps_point(n),
+				  inv_e_metric_(n, n),
+				  inv_e_metric_llt_matrixU_(n, n) {
     inv_e_metric_.setIdentity();
+    inv_e_metric_llt_matrixU_.setIdentity();
   }
 
   /**
-   * Set elements of mass matrix
+   * Set inverse metric
    *
    * @param inv_e_metric initial mass matrix
    */
-  void set_metric(const Eigen::MatrixXd& inv_e_metric) {
+  void set_inv_metric(const Eigen::MatrixXd& inv_e_metric) {
```

Comment:
I think the API is supposed to be just `set_metric` for each point no?

Reply:
Well it's the inverse metric, not the metric, so I just changed the names everywhere.